py-nominal-streaming/src/lazy_dataset_stream_builder.rs

```diff
@@ -0,0 +1,91 @@
+use std::path::PathBuf;
+
+use anyhow::anyhow;
+use anyhow::Result;
+use nominal_streaming::prelude::*;
+use nominal_streaming::stream::NominalDatasetStreamBuilder;
+
+use crate::nominal_stream_opts::NominalStreamOptsWrapper;
+
+#[derive(Clone)]
+pub struct CoreTarget {
+    pub token: BearerToken,
+    pub rid: ResourceIdentifier,
+}
+
+#[derive(Clone)]
+pub struct FileTarget {
+    pub path: PathBuf,
+}
+
+#[derive(Clone, Default)]
+pub struct StreamTargets {
+    pub core_target: Option<CoreTarget>,
+    pub file_target: Option<FileTarget>,
+    pub file_fallback: Option<PathBuf>,
+}
+
+/// Simple wrapper around a NominalDatasetStreamBuilder that allows for lazily configuring
+/// streaming to core vs. files without a tokio runtime readily available.
+#[derive(Clone, Default)]
+pub struct LazyDatasetStreamBuilder {
```

Comment:
seems very reasonable to make the existing NominalDatasetStreamBuilder more flexible - feel free to add this functionality there in a future PR so there's less to manage